tutorials/cloudbuild-angular-universal/index.md

```diff
@@ -0,0 +1,301 @@
+```
+git add package.json
+```
+9. Commit your changes to the git repo
+```
+git commit -m "pregenerate changes"
+```
+### Configure a Cloud Storage Bucket and Cloud Load Balancer to host your Angular Application on Cloud CDN
+
+1. Create the content cloud storage bucket.
+```
+gsutil mb gs://$PROJECT-angular-app
+```
+2. Create the backend-bucket.
+```
+gcloud compute backend-buckets create $PROJECT-angular-app-backend \
+--gcs-bucket-name=$PROJECT-angular-app \
+--enable-cdn
+```
+3. Create a multi-regional ip address.
+```
+gcloud compute addresses create angular-app-ip --global
+ANGULAR_APP_IP=$(gcloud compute addresses list  --filter="name=angular-app-ip" --format="value(address)")
+```
+3. Create the url map.
+```
+gcloud compute url-maps create web-map --default-backend-bucket $PROJECT-angular-app-backend
+```
+4. Create the http proxy.
+```
+gcloud compute target-http-proxies create http-lb-proxy \
+--url-map web-map
+```
+5. Create the forwarding rule.
+```
+gcloud compute forwarding-rules create http-content-rule \
+--address angular-app-ip \
+--global \
+--target-http-proxy http-lb-proxy \
+--ports 80
+```
+## Create the Cloudbuild file and add it to the git repsoitory
+1. Give the cloudbuild cloud storage admin access.
+
+```bash
+CLOUD_BUILD_ACCOUNT=$(gcloud projects get-iam-policy $PROJECT --filter="(bindings.role:roles/cloudbuild)"  --flatten="bindings[].members" --format="value(bindings.members[])")
+gcloud projects add-iam-policy-binding $PROJECT   --member $CLOUD_BUILD_ACCOUNT  --role roles/storage.admin
+```
+2. Create the ```cloudbuild.yaml``` file.
+
+```bash
+cat <<CLOUDBUILD_FILE>cloudbuild.yaml
+steps:
+- id: install_packages
+  name: 'gcr.io/cloud-builders/npm'
+  args:
+  - 'install'
+- id: prerender_browser_files
+  name: 'gcr.io/cloud-builders/npm'
+  args:
+  - 'run'
+  - 'build:prerender'
+  waitFor:
+  - install_packages
+- id: copy_prerendered_files
+  name: 'gcr.io/cloud-builders/gsutil'
+  args: ['cp','-r','dist/browser/*', '\${_ANGULAR_APP_BUCKET_PATH}']
+  waitFor:
+  - prerender_browser_files
+- id: set_website_configuration
+  name: 'gcr.io/cloud-builders/gsutil'
+  args: ['web', 'set', '-m', 'index.html','\${_ANGULAR_APP_BUCKET_PATH}']
+  waitFor:
+  - copy_prerendered_files
+- id: set_permissions_for_website_files
+  name: 'gcr.io/cloud-builders/gsutil'
+  args: ['acl','ch','-u','AllUsers:R','-r', '\${_ANGULAR_APP_BUCKET_PATH}']
+  waitFor:
+  - copy_prerendered_files
+CLOUDBUILD_FILE
+```
+
+3. Add and commit ```cloudbuild.yaml``` to the git repository.
+```
+git add cloudbuild.yaml && git commit -m "add cloudbuild.yaml"
+```
+### Create a Build Trigger that will build, test and deploy your application to the Cloud CDN
+You can create a trigger on the [build triggers page](https://console.cloud.google.com/cloud-build/triggers) of the GCP Console by following these steps:
+
+1. Click **Create Trigger** OR **Add Trigger** button
+1. Select **Cloud Source Repository** Click **Continue**.
```

Comment:
'Select **Cloud Source Repository** Click **Continue**.'
->
'Select **Cloud Source Repository** and click **Continue**.'